meson.build

```diff
@@ -67,12 +67,27 @@ endif
 # new enough, if not use a static-pic-lib regardless of
 # default-library.
 libetpan_config = find_program('libetpan-config', required: false)
-if (not get_option('monolith')
-    and not get_option('force-etpan-fallback')
-    and libetpan_config.found()
-    and run_command(libetpan_config, ['--version']).stdout().strip()
-                   .version_compare('>=1.8'))
+libetpan_fallback = (get_option('monolith')
+                     or get_option('force-etpan-fallback')
+                     or not libetpan_config.found())
+
+# If we want to use a found etpan version, we have to make sure it's compatible.
+# Sadly libetpan doesn't use the same version accross all platforms, so we have
+# to check this depending on the platform.
+if libetpan_fallback == false
   etpan_version = run_command(libetpan_config, ['--version']).stdout().strip()
+  message('Dependency libetpan found: @0@. Checking for compatibility.'.format(etpan_version))
+  if (etpan_version.version_compare('>=1.8')
+      or host_machine.system() == 'darwin'
+      and etpan_version.version_compare('>=1.6'))
+    message('Check done. Installed libetpan version is compatible.')
+    libetpan_is_compatible = true
+  else
+    error('Check done. Installed libetpan version is NOT compatible. You need at least 1.9 for most platforms or 1.6 for Darwin/Mac')
```

Comment:
I think the windows version is 1.6 too

Reply:
Oh good to know. But i would prefer if we do this if we try to make it work on windows. Want to focus on mac in this pr.